lib/StackageToHackage/Hackage.hs

```diff
@@ -111,3 +129,33 @@ genFreeze Resolver{deps, flags} ignore =
         pick (SourceDep _) = Nothing
         pick (LocalDep _) = Nothing
         noSelfs (pkgName -> n) = notElem n ignore
+
+
+-- | Acquire all package identifiers from a list of subdirs
+-- of a git repository.
+getPackageIdent :: Git -> IO [PackageIdentifier]
+getPackageIdent (Git (T.unpack -> repo) (T.unpack -> commit) (fmap T.unpack -> subdirs)) =
+  withSystemTempDirectory "stack2cabal" $ \dir -> do
```

Comment:
We clone into a temp dir and reset to the "offending" commit, then use cabal to parse the .cabal file and get `PackageIdentifier`. This is semi-expensive, hence optional from the cli. It would be cool to re-use the repos cloned by cabal already, but that would probably require far more work.